### PR TITLE
Transfer call e2e tests

### PR DESCRIPTION
**Pull request by sinapsist**
# 🏗️ PR ➾

Fixes # (issue)

## 🪂 Pre-Merge Checklist

- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x] 🦀 Automated tests have been written and added to this PR
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [x] 🔱 The test plan has been implemented and verified by an SDET
- [x] 🏄‍♂️ Myself and others have built this branch and done manual testing on the change
- [x] ⛵ A summary describing the change has been written for release notes

----------------------------------------------------------------------------------------------------------------------------

## 🪁 Description

Please include a summary of the change and add the issue number this is related to above. Please also include relevant motivation and context for the change
(e.g. what was the existing behavior, why did it break, etc.)

## 🎢 Test Plan

Please describe the testing strategy for this change

----------------------------------------------------------------------------------------------------------------------------

### 🛸 Type of Change

- [ ] 🐟 Minor change (non-breaking change of very limited scope)
- [ ] 🦑 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🌊 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🐳 Major refactor (breaking changes and significant impact to the codebase)

----------------------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts [hunk 1]
```diff
@@ -100,6 +107,130 @@ describe('E2E Testing for taqueria taquito plugin', () => {
...
+			expect(transferResult.stderr).toContain('Error during transfer operation');
+			expect(transferResult.stderr).toContain('TezosOperationError: (temporary) proto.alpha.contract.balance_too_low');
+			expect(transferResult.stderr).toContain('No operations performed');
+		} catch (error) {
+			throw new Error(`error: ${error}`);
+		}
+	});
+
+	// TODO: NEED TO COMPLETE THIS TEST
+	test.skip('Verify that taqueria taquito plugin can transfer amount of tezos using call command from one account to another', async () => {
```
Comment (alexzbusko) [inline tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts:165]: For skipped tests I would remove them from the PR and then open a ticket with the tests in them that need to be completed

File: tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts [hunk 2]
```diff
@@ -100,6 +107,130 @@ describe('E2E Testing for taqueria taquito plugin', () => {
...
 	});
 
+	test('Verify that taqueria taquito plugin can transfer amount of tezos using transfer command from one account to another', async () => {
+		try {
+			// Setting up docker container name
+			dockerName = 'local';
+			environment = 'development';
+
+			// 1. Start local sandbox
+			await exec(`taq start sandbox`, { cwd: `./${taqueriaProjectPath}` });
```
Comment (alexzbusko) [inline tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts:117]: You don't need to start a sandbox here, one is already getting started in the beforeAll clause above

**Comment by alexzbusko** [comment]
@sinapsist I would update this PR with the main stuff and that should resolve the mac runner errors.

There are also some test cases that I think could go into the ticket for the currently skipped tickets:
- Checking there is an error for sending 0 tez in a transfer
- Checking that a transfer to an unknown address gives an error
- If you send a transfer from an account to itself it should only burn gas. I don't know if this should be allowed in general
- Trying to send an amount smaller than a minimum size should give an error


One question I have is it only seems to transfer from the default account. Is there any way to transfer from any other account or is it stuck to that one?

File: tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts (hunk 1 above)
Comment (sinapsist) [reply tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts]: I cleaned up all skipped tests for transfer/call command.

Comment (sinapsist) [reply tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts]: Yes, call is an alias - https://taqueria.io/docs/plugins/plugin-taquito/

File: tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts (hunk 2 above)
Comment (sinapsist) [reply tests/e2e/taqueria-plugin-taquito-flextesa.spec.ts]: Yeah, I removed it.

**Review by GImbrailo — APPROVED** [review]
LGTM

**Review by AlirezaHaghshenas — APPROVED** [review]

**Comment by sinapsist** [comment]
@alexzbusko 
I merged the main, and it still failing on mac runner.

There are also some test cases that I think could go into the ticket for the currently skipped tickets:
Checking if there is an error for sending 0 tez in a transfer

Sending 0 tez is not an invalid case if you don't specify --tez it will send 0 (see documentation)

Checking that a transfer to an unknown address gives an error
Sure I can add it in a separate PR.

If you send a transfer from an account to itself it should only burn gas. I don't know if this should be allowed in general
It will also burn only gas if you send 0 tez - it is covered 

Trying to send an amount smaller than a minimum size should give an error
@alexzbusko Can you provide more details, you mean less than 0 mtez? But again, even 0 tez is a valid scenario.

One question I have is it only seems to transfer from the default account. Is there any way to transfer from any other account, or is it stuck to that one?
It can only transfer from the default account to another account or to the contract.

**Comment by sinapsist** [comment]
@alexzbusko  
I will also add transfering negative tez amount to the negative tests into next PR.

**Pull request merged**